### PR TITLE
Added a more strict join criteria for employers + pagination

### PR DESCRIPTION
**Pull request by mauriciofreshw**
This will reduce the output combination while doing the join and greatly reduce the decompose function load.

### CODE REVIEW

**Review by ashtonmeuser — APPROVED** [review]

File: server/services/participants.js [hunk 1]
```diff
@@ -89,7 +89,16 @@ const getParticipants = async (user) => {
...
+        on: user.isHA || user.isEmployer
+          ? {
+            participant_id: 'id',
+            current: true,
+            employer_id: user.id,
+          }
+          : {
+            participant_id: 'id',
+            current: true,
+          },
```
Comment (ashtonmeuser) [inline server/services/participants.js:101]: Check out this slick move
```suggestion
        on: {
            participant_id: 'id',
            current: true,
            ...(user.isHA || user.isEmployer) && {employer_id: user.id},
          },
```

**Review by ashtonmeuser — APPROVED** [review]

**Review by Zenen-freshworks — DISMISSED** [review]

File: server/services/participants.js [hunk 2]
```diff
@@ -81,50 +81,90 @@ const decomposeParticipantStatus = (raw) => {
...
-    ? await table.findDoc(criteria)
+    ? await table.findDoc(criteria, options)
     : [];
 
   if (showStatus) {
     participants = decomposeParticipantStatus(participants);
   }
 
+  const paginationData = pagination && {
+    lastId: participants.length > 0 && participants[participants.length - 1][pagination.field || 'id'],
```
Comment (ashtonmeuser) [inline server/services/participants.js:138]: Won't `lastId` end up being a boolean?

**Review by Zenen-freshworks — DISMISSED** [review]

File: server/services/participants.js [hunk 3]
```diff
@@ -41,90 +41,138 @@ const setParticipantStatus = async (
...
+  criteria = {
+    ...criteria,
+    ...fsaFilter && { 'body.postalCodeFsa ilike': `${fsaFilter}%` },
+  };
 
-const getParticipants = async (user) => {
-  const criteria = user.isSuperUser || user.isMoH ? {} : userRegionQuery(user.regions, 'preferredLocation');
   let table = dbClient.db[collections.PARTICIPANTS];
-  const showStatus = user.isHA || user.isEmployer || user.isSuperUser;
+  const join1Name = 'join1';
```
[the remark this answers is not in the document]
Comment (mauriciofreshw) [reply server/services/participants.js:84]: Yep!

**Review by Zenen-freshworks — DISMISSED** [review]

**Review by Zenen-freshworks — APPROVED** [review]

File: server/services/participants.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ashtonmeuser) [reply server/services/participants.js:84]: I'm also a little turned off by the `join1` name, but just a nitpick.

**Review by ashtonmeuser — APPROVED** [review]

**Pull request merged**